### PR TITLE
employees: combine put/post into update

### PR DESCRIPTION
**Pull request by kenziebottoms**
# Description
`PUT` and `POST` now use the same model and controller functions.

## Related Ticket(s)
#103 

## Steps to Test Solution
1. `npm run db:generate`
1. Depending on whether or not Joe's PR #117 has been merged, comment out superfluous routes
1. `npm start`
1. Make a `POST` request to route root (`/employees`, `/customers`, etc.)
1. Make a `PUT` request to an existing ID
1. Make a `PUT` request to an ID that doesn't have an object (use id 100 because you know that won't be in use)
1. Check that all requests do what you expect them to.
1. All requests should return the ID of the updated/created object

### CODE REVIEW

File: models/employees/EmployeesModel.js [hunk 1]
```diff
@@ -42,11 +26,12 @@ module.exports.putEmployee = (id, first_name, last_name, department_id) =>
         department_id
       )
       VALUES (
-        ${id},
+        ${id == undefined ? null : id},
```
Comment (joechesney) [inline models/employees/EmployeesModel.js:29]: double nice

File: controllers/employees/employeesCtrl.js [hunk 2]
```diff
@@ -5,38 +5,26 @@ const appRoot = process.cwd();
...
-  }
-};
 
 module.exports.updateEmployee = (req, res, next) => {
   let { first_name, last_name, department_id } = req.body;
-  if (first_name, last_name, department_id) {
-    putEmployee(req.params.id, first_name, last_name, department_id)
-    .then(resp => res.status(201).json(resp))
-    .catch(err => next(err))
+  if (first_name && last_name && department_id) {
```
Comment (joechesney) [inline controllers/employees/employeesCtrl.js:23]: nice

File: models/employees/EmployeesModel.js [hunk 3]
```diff
@@ -42,11 +26,12 @@ module.exports.putEmployee = (id, first_name, last_name, department_id) =>
...
         "${first_name}",
         "${last_name}",
         "${department_id}"
-      )`,
-      err => err ? reject(err) : resolve(id)
+      )`, function(err) {
+        return err ? reject(err) : resolve(this.lastID)
```
Comment (joechesney) [inline models/employees/EmployeesModel.js:34]: triple nice

**Comment by jordan-castelloe** [comment]
testing!

**Review by joechesney — APPROVED** [review]
I did not test this, but the code looks great

**Review by jordan-castelloe — APPROVED** [review]
Looks good!

**Pull request merged**